stock_picking_group_by_partner_by_carrier/models/stock_picking.py

```diff
@@ -179,11 +180,12 @@ def get_delivery_report_lines(self):
                 sales_and_moves = self.env["stock.move.line"]
                 fake_record["product_uom_id"] = fake_record.pop("product_uom")
                 for sale, sale_moves in grouped_moves:
-                    line_desc = sale.get_name_for_delivery_line()
-                    fake_record.update(
-                        {"description_picking": line_desc, "origin": sale.name}
-                    )
-                    sales_and_moves |= sales_and_moves.new(fake_record.copy())
+                    if sale:
+                        line_desc = sale.get_name_for_delivery_line()
+                        fake_record.update(
+                            {"description_picking": line_desc, "origin": sale.name}
```

Comment:
hmm we'll miss the keys. Is there a check or is going to fail in the report?

Reply:
This fake record dict is only used to append a section line with the name of the sale order, if there is no order we don't want to create such line.